app/forms/providers/bank_statement_form.rb

```diff
@@ -0,0 +1,155 @@
+module Providers
+  class BankStatementForm
+    include ActiveModel::Model
+    include ActiveModel::Validations::Callbacks
+    include MalwareScanning
+
+    ATTACHMENT_TYPE = "bank_statement_evidence".freeze
+    ATTACHMENT_TYPE_CAPTURE = /^#{ATTACHMENT_TYPE}_(\d+)$/
+
+    MAX_FILE_SIZE = 7.megabytes
+
+    ALLOWED_CONTENT_TYPES = %w[
+      application/pdf
+      application/msword
+      application/vnd.oasis.opendocument.text
+      text/rtf
+      text/plain
+      application/rtf
+      image/jpeg
+      image/png
+      image/tiff
+      image/bmp
+      image/x-bitmap
+    ].freeze
+
+    WORD_DOCUMENT = "application/vnd.openxmlformats-officedocument.wordprocessingml.document".freeze
+
+    attr_accessor :original_file,
+                  :original_filename,
+                  :legal_aid_application_id,
+                  :upload_button_pressed
+
+    validate :at_least_one_file_or_draft
+
+    def self.max_file_size
+      MAX_FILE_SIZE
+    end
+
+    # rubocop:disable Rails/SaveBang
+    def save_as_draft
+      @draft = true
+      save
+    end
+    # rubocop:enable Rails/SaveBang
+
+    # Files already uploaded and created with bank statement associations
+    # so we do not need to save anything at this point, only to validate that
+    # there is one or more bank statements or the application is draft
+    def save
+      valid?
+    end
+
+    def upload
+      return if original_file.nil?
+
+      validate_original_file
+      create_attachment(original_file) if errors.blank?
+    end
+
+    def legal_aid_application
+      @legal_aid_application ||= LegalAidApplication.find(legal_aid_application_id)
+    end
+
+    def provider_uploader
+      legal_aid_application.provider
+    end
+
+  private
+
+    def validate_original_file
+      self.original_filename = original_file.original_filename
+      scanner_down
+      malware_scan
+      file_empty
+      disallowed_content_type
+      too_big
+    end
+
+    def at_least_one_file_or_draft
+      return if any_bank_statements_or_draft?
+
+      @errors.add(:original_file, :no_file_chosen)
+    end
+
+    def any_bank_statements_or_draft?
+      legal_aid_application.attachments.bank_statement_evidence.any? || draft?
+    end
+
+    def draft?
+      @draft
+    end
+
+    # can be shared with v1 bank statement controller
```

Comment:
Are these reminders for future tickets? If so, and the ticket has been written, could we add a ticket number here